### PR TITLE
Feature/650 flyby guidance refactor

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-650
* **Review:** By commit 
* **Merge strategy:** Merge (no squash)  

## Description
Refactor the flybyPointing module to use only Cpp. Refactor adds setters and getters, and performs several small code improvements

## Verification
Test was changed to use setters and getters, but still passes unchanged

## Documentation
None

## Future work
None

### CODE REVIEW

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp [hunk 1]
```diff
@@ -116,18 +111,15 @@ void FlybyPoint::UpdateState(uint64_t CurrentSimNanos)
         }
 
         /*! compute inertial-to-reference DCM at time of read */
-        for (int i=0; i<3; i++) {
-            this->R0N[0][i] = ur_N[i];
-            this->R0N[1][i] = ut_N[i];
-            this->R0N[2][i] = uh_N[i];
-        }
+        this->R0N.row(0) = ur_N;
```
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp:91]: :large_orange_diamond: Looks like this change snuck in here from he subsequent commit?

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.h [hunk 2]
```diff
@@ -49,18 +49,18 @@ class FlybyPoint: public SysModel {
...
     Message<AttRefMsgPayload> attRefOutMsg;                     //!< Attitude reference output message
     AttRefMsg_C attRefOutMsgC = {};                             //!< C-wrapped attitude reference output message
 
 private:
+    double     timeBetweenFilterData = 0;       //!< time between two subsequent reads of the filter information
+    double     toleranceForCollinearity = 0;            //!< tolerance for singular conditions when position and velocity are collinear
+    int64_t    signOfOrbitNormalFrameVector = 1;  //!< Sign of orbit normal vector to complete reference frame
+    FlybyModel chosenFlybyModel = rectilinear;              //!< enum to indicate which flyby model is being used
+
     bool            firstRead;           //!< variable to attest if this is the first read after a Reset
```
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.h]: :large_orange_diamond: We should class initialize these also. E.g. 
`bool firstRead=true;` and so on

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp [hunk 3]
```diff
@@ -71,44 +72,28 @@ void FlybyPoint::UpdateState(uint64_t CurrentSimNanos)
...
-            // compute flight path angle at the time of read
-            this->gamma0 = atan(v3Dot(relativeState.v_BN_N, ur_N) / v3Dot(relativeState.v_BN_N, ut_N));
-            this->singularityFlag = nonSingular;
-        }
+        Eigen::Vector3d ur_N = this->r_BN_N / this->r_BN_N.norm();
+        Eigen::Vector3d uv_N = this->v_BN_N / this->v_BN_N.norm();
+        /*! assert r and v are not collinear (collision trajectory) */
+        assert(std:abs(1 - ur_N.dot(uv_N)) < this->toleranceForCollinearity);
+
+        Eigen::Vector3d uh_N;
```
Comment (patkenneally) [inline src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp]: :large_orange_diamond: Is there a preference for declaring and assigning the variable on different lines? I don't know if there's a benefit either way. Just curious.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp (hunk 1 above)
Comment (thibaudteil) [reply src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp:91]: Not sure I understood this one

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.h (hunk 2 above)
Comment (thibaudteil) [reply src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.h]: done

File: src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp (hunk 3 above)
Comment (thibaudteil) [reply src/fswAlgorithms/attGuidance/flybyPoint/flybyPoint.cpp]: improved this too

**Pull request merged**